packages/plugin-react-refresh/index.js

```diff
@@ -37,6 +37,10 @@ window.__vite_plugin_react_preamble_installed__ = true
 function reactRefreshPlugin(opts) {
   let shouldSkip = false
   let base = '/'
+  const filter = createFilter(
+    (opts && opts.include) || /\.(?:tsx|jsx|js)$/,
```

Comment:
Why are we changing this regex? It was `/\.(t|j)sx?$/` before. The new includes `'.js'` files. The default in Vite for react is that `.js` files are not considered to include JSX (users should use `.jsx`).

And no worries about typos, we all do the same. That is why it is good that there are several people reviewing :)

Reply:
The old version included `.js`, `.jsx`, `.ts`, and `.tsx`. It's not possible to include react components in `.ts` files, so I removed it.  Is there no way to configure vite to allow react components inside of `.js` files, as well?

Reply:
Oh, sorry, I missed the `?`, nvm

Reply:
Hmmmm, good point.  It's technically possible to create react components without JSX using `React.createElement`.  I'm not certain those are valid fast-refresh targets, but maybe?  To be safe I'll change back to the original pattern.